File: internal/util/apiclient/conversion/env.go
// Copyright 2024 Daytona Platforms Inc.
// SPDX-License-Identifier: Apache-2.0

package conversion

import "github.com/daytonaio/daytona/pkg/apiclient"

func ToEnvVarsMap(envVars []apiclient.EnvironmentVariable) map[string]string {
	envVarsMap := map[string]string{}

	for _, envVar := range envVars {
		envVarsMap[envVar.Key] = envVar.Value
	}

	return envVarsMap
}

File: apps/dashboard/src/lib/env.ts
/*
 * Copyright Daytona Platforms Inc.
 * SPDX-License-Identifier: AGPL-3.0
 */

import { parse } from 'dotenv'

export interface EnvVar {
  key: string
  value: string
}

export function parseEnvFile(src: string): EnvVar[] {
  const parsed = parse(src)
  return Object.entries(parsed).map(([key, value]) => ({ key, value }))
}
